server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeQueryHelper.java

```diff
@@ -152,11 +163,38 @@ private static MetricStat validateStarTreeMetricSupport(
             MetricStat metricStat = ((MetricAggregatorFactory) aggregatorFactory).getMetricStat();
             field = ((MetricAggregatorFactory) aggregatorFactory).getField();
 
-            if (supportedMetrics.containsKey(field) && supportedMetrics.get(field).contains(metricStat)) {
-                return metricStat;
-            }
+            return supportedMetrics.containsKey(field) && supportedMetrics.get(field).contains(metricStat);
+        }
+        return false;
+    }
+
+    private static boolean validateDateHistogramSupport(
+        CompositeDataCubeFieldType compositeIndexFieldInfo,
+        AggregatorFactory aggregatorFactory
+    ) {
+        if ((aggregatorFactory instanceof DateHistogramAggregatorFactory) == false
+            && aggregatorFactory.getSubFactories().getFactories().length != 1
+            && (compositeIndexFieldInfo.getDimensions().get(0) instanceof DateDimension) == false) {
+            return false;
+        }
+
+        DateHistogramAggregatorFactory dateHistogramAggregatorFactory = (DateHistogramAggregatorFactory) aggregatorFactory;
+        // date fields are indexed at top of tree
+        DateDimension starTreeDateDimension = (DateDimension) compositeIndexFieldInfo.getDimensions().get(0);
```

Comment:
This could become problematic if we change the order in future. Can we rather try finding the dimension based on id / name.